Samples/WinMLSamplesGallery/WinMLSamplesGalleryNative/AdapterList.cpp

```diff
@@ -0,0 +1,91 @@
+#include "pch.h"
+#include "AdapterList.h"
+#include "AdapterList.g.cpp"
+#include <initguid.h>
+#include <dxcore.h>
+#include "d3d12.h"
+#include "winrt/Microsoft.AI.MachineLearning.h"
+#include "Microsoft.AI.MachineLearning.Native.h"
+
+namespace winrt::WinMLSamplesGalleryNative::implementation
+{
+    winrt::com_array<hstring> AdapterList::GetAdapters() {
+        // Create an adapter factory
+        winrt::com_ptr<IDXCoreAdapterFactory> adapterFactory;
+        winrt::check_hresult(::DXCoreCreateAdapterFactory(adapterFactory.put()));
+
+        // Retrieve an adapter list
+        winrt::com_ptr<IDXCoreAdapterList> d3D12CoreComputeAdapters;
+        GUID attributes[]{ DXCORE_ADAPTER_ATTRIBUTE_D3D12_CORE_COMPUTE };
+        winrt::check_hresult(
+            adapterFactory->CreateAdapterList(_countof(attributes),
+                attributes,
+                d3D12CoreComputeAdapters.put()));
+
+        // Get the driver descriptions from the adapters in the list
+        const uint32_t count{ d3D12CoreComputeAdapters->GetAdapterCount() };
+        com_array<hstring> driver_descriptions = com_array<hstring>(count);
+        for (uint32_t i = 0; i < count; ++i)
+        {
+            winrt::com_ptr<IDXCoreAdapter> candidateAdapter;
+            winrt::check_hresult(
+                d3D12CoreComputeAdapters->GetAdapter(i, candidateAdapter.put()));
+            CHAR description[128];
+            candidateAdapter->GetProperty(DXCoreAdapterProperty::DriverDescription, sizeof(description), description);
+            driver_descriptions[i] = to_hstring(description);
+        }
+
+        return driver_descriptions;
+    }
+
+    winrt::Microsoft::AI::MachineLearning::LearningModelDevice AdapterList::CreateLearningModelDeviceFromAdapter(winrt::hstring description) {
+        // Create an adapter factory
+        winrt::com_ptr<IDXCoreAdapterFactory> adapterFactory;
+        winrt::check_hresult(::DXCoreCreateAdapterFactory(adapterFactory.put()));
+
+        // Retrieve an adapter list
+        winrt::com_ptr<IDXCoreAdapterList> d3D12CoreComputeAdapters;
+        GUID attributes[]{ DXCORE_ADAPTER_ATTRIBUTE_D3D12_CORE_COMPUTE };
+        winrt::check_hresult(
+            adapterFactory->CreateAdapterList(_countof(attributes),
+                attributes,
+                d3D12CoreComputeAdapters.put()));
+
+        // Find the driver in the list with the matching description
```

Comment:
I think should be adapter instead of driver in the comment

Reply:
Fixed.